server/resp/src/main/java/org/infinispan/server/resp/commands/connection/MEMORY.java

```diff
@@ -0,0 +1,108 @@
+package org.infinispan.server.resp.commands.connection;
+
+import static org.infinispan.server.resp.RespConstants.CRLF_STRING;
+
+import java.nio.charset.StandardCharsets;
+import java.util.List;
+import java.util.concurrent.CompletionStage;
+
+import org.infinispan.security.AuthorizationPermission;
+import org.infinispan.server.resp.ByteBufferUtils;
+import org.infinispan.server.resp.Consumers;
+import org.infinispan.server.resp.Resp3Handler;
+import org.infinispan.server.resp.RespCommand;
+import org.infinispan.server.resp.RespErrorUtil;
+import org.infinispan.server.resp.RespRequestHandler;
+import org.infinispan.server.resp.commands.Resp3Command;
+
+import io.netty.channel.ChannelHandlerContext;
+
+/**
+ * <a href="https://redis.io/commands/memory/">MEMORY</a>
+ *
+ * @since 15.0
+ */
+public class MEMORY extends RespCommand implements Resp3Command {
+   public MEMORY() {
+      super(-2, 0, 0, 0);
+   }
+
+   @Override
+   public CompletionStage<RespRequestHandler> perform(Resp3Handler handler, ChannelHandlerContext ctx,
+                                                      List<byte[]> arguments) {
+      handler.checkPermission(AuthorizationPermission.ADMIN);
+      String subcommand = new String(arguments.get(0), StandardCharsets.UTF_8).toUpperCase();
```

Comment:
I believe all commands have to be US_ASCII afaik.

Reply:
Done